src/Microsoft.IdentityModel.Tokens.Saml/SamlTokenUtilities.cs

```diff
@@ -102,5 +100,62 @@ internal static IEnumerable<SecurityKey> GetKeysForTokenSignatureValidation(stri
             }
             return null;
         }
+
+        /// <summary>
+        /// Creates <see cref="Claim"/>'s from <paramref name="claimsCollection"/>.
+        /// </summary>
+        /// <param name="claimsCollection"> A dictionary that represents a set of claims.</param>
+        /// <returns> A collection of <see cref="Claim"/>'s created from the <paramref name="claimsCollection"/>.</returns>
+        internal static IEnumerable<Claim> CreateClaimsFromDictionary(IDictionary<string, object> claimsCollection)
+        {
+            List<Claim> claims = null;
+            if (claimsCollection == null)
+                return claims;
+
+            claims = new List<Claim>();
+            foreach (var claim in claimsCollection)
+            {
+                string claimType = claim.Key;
+                object claimValue = claim.Value;
+
+                if (claimValue != null)
+                {
+                    string valueType = TokenUtilities.GetClaimValueTypeFromValue(claimValue);
+
+                    if (claimValue.GetType().Name == typeof(List<>).Name)
+                    {
+                        foreach (var item in (IList)claimValue)
+                            claims.Add(new Claim(claimType, item.ToString(), valueType));
+                    }
+                    else
+                    {
+                        claims.Add(new Claim(claimType, claimValue.ToString(), valueType));
+                    }
+                }
+            }
+
+            return claims;
+        }
+
+        /// <summary>
+        /// Merges <paramref name="claims"/> and <paramref name="subjectClaims"/>
+        /// </summary>
+        /// <param name="claims"> A dictionary of claims.</param>
+        /// <param name="subjectClaims"> A collection of <see cref="Claim"/>'s</param>
+        /// <returns> A merged list of <see cref="Claim"/>'s.</returns>
+        internal static IEnumerable<Claim> GetAllClaims(IDictionary<string, object> claims, IEnumerable<Claim> subjectClaims)
+        {
+            IEnumerable<Claim> allClaims = null;
```

Comment:
It is worth consider if this should ever be allowed to return null.

Null doesn't appear to have much meaning to callers, only if there are items in the returned collection. We can get close to a Null Object Pattern with just returning an empty list.